src/next-transpile-modules.js

```diff
@@ -21,44 +25,45 @@ const regexEqual = (x, y) => {
   );
 };
 
-const generateIncludes = (modules) => {
-  return [
-    new RegExp(`(${modules.map(safePath).join('|')})$`),
-    new RegExp(`(${modules.map(safePath).join('|')})${PATH_DELIMITER}(?!.*node_modules)`)
-  ];
-};
-
-const generateExcludes = (modules) => {
-  return [
-    new RegExp(
-      `node_modules${PATH_DELIMITER}(?!(${modules.map(safePath).join('|')})(${PATH_DELIMITER}|$)(?!.*node_modules))`
-    )
-  ];
-};
-
 /**
- * On Windows, the Regex won't match as Webpack tries to resolve the
- * paths of the modules. So we need to check for \\ and /
+ * Resolve modules to their real paths
+ * @param {string[]} modules
  */
-const safePath = (module) => module.split(/[\\\/]/g).join(PATH_DELIMITER);
+const generateResolvedModules = (modules) => {
```

Comment:
Fixes issues where. the package is unable to resolve in monorepo directories



```js
const generateResolvedModules = (modules) => {
  const resolvedModules = modules
    .map((module) => {
      let resolved;
      try {
        resolved = resolve.sync(module);
      } catch (e) {
         resolved = require.main.paths.find((resolutionPath) => {
          return fs.existsSync(path.join(resolutionPath, module));
        });
      }

      if (!resolved)
        throw new Error(
          `next-transpile-modules: could not resolve module "${module}". Are you sure the name of the module you are trying to transpile is correct?`
        );

      return resolved;
    })
    .map(path.dirname);

  return resolvedModules;
};
```

Reply:
I did not have issues with monorepos (cf. yarn workspaces tests passing), can you elaborate?

Reply:
when i run resole sync, it would give me cannot find module errors. 

Do you have a beta pushed to npm of this PR? i copied it and it did not resolve